daffodil-test/src/test/scala/org/apache/daffodil/section13/validation_errors/PadCharacter.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.section13.validation_errors
+
+import org.apache.daffodil.tdml.DFDLTestSuite
+import org.apache.daffodil.xml.XMLUtils
+import org.junit.Test
+
+class PadCharacter {
+
+  val tdml = XMLUtils.TDML_NAMESPACE
+  val dfdl = XMLUtils.DFDL_NAMESPACE
+  val xsd = XMLUtils.XSD_NAMESPACE
+  val tns = XMLUtils.EXAMPLE_NAMESPACE
+
+  @Test def test_short_form_pad_char() = {
+    // This test demonstrates that you cannot use a literal whitespace character
+    // in the short form of the pad character's property binding syntax
+    val testSuite =
+      <tdml:testSuite suiteName="PadCharTests" description="Section 13 - Pad Character Tests"
+                      xmlns:tdml={ tdml } xmlns:dfdl={ dfdl } xmlns:xs={ xsd } xmlns:tns={ tns }>
+        <tdml:defineSchema name="padCharSchema">
+          <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd" />
+          <dfdl:format ref="tns:GeneralFormat" textTrimKind="padChar" textPadKind="padChar" />
+          <xs:element name="shortForm" type="xs:string" dfdl:textStringPadCharacter=" " />
+        </tdml:defineSchema>
+
+        <tdml:parserTestCase name="short_form_pad_char" root="shortForm" model="padCharSchema">
+          <tdml:document>
+            <tdml:documentPart type="text"><![CDATA[      hello]]></tdml:documentPart>
+          </tdml:document>
+          <tdml:errors>
+            <tdml:error>Value ' ' is not facet-valid</tdml:error>
```

Comment:
I would make this <tdml:error>facet</tdml:error><tdml:error>valid</tdml:error><tdml:error>NonEmptyStringLiteral</tdml:error> so that it's not sensitive to wording or quoting conventions. 

Same idea applies to your other tests.